pkg/events/correlation_context.go

```diff
@@ -33,9 +33,28 @@ func ContextWithCorrelationID(ctx context.Context, cids ...string) context.Conte
 	if !ok {
 		return context.WithValue(ctx, correlationKey{}, cids)
 	}
+	merged := mergeCorrelationIDs(existing, cids)
+	return context.WithValue(ctx, correlationKey{}, merged)
+}
+
+// CorrelationIDsFromContext returns the correlation IDs that are attached to the context.
+func CorrelationIDsFromContext(ctx context.Context) []string {
+	cids, ok := ctx.Value(correlationKey{}).([]string)
+	if !ok {
+		return nil
+	}
+	return cids
+}
+
+// NewCorrelationID returns a new random correlation ID.
+func NewCorrelationID() string {
+	return ulid.MustNew(ulid.Now(), rand.Reader).String()
+}
 
-	// Since existing and cids are sorted, merging them will result in a sorted slice.
-	// See https://en.wikipedia.org/wiki/Merge_sort
+// mergeCorrelationIDs returns sorted array of correlation ids from context and payload
```

Comment:
1. This function returns a slice, not an array
2. This function merges 2 sorted string slices, it has nothing to do with correlation IDs, context or payload